docs/GettingStartedDocs/OP-TEE/Hardware/ScalysTrustBox.md

```diff
@@ -0,0 +1,393 @@
+
+```bash
+make firmware HAB=1
+```
+
+**Note:** Do not attempt to use `-j`. The steps that are parallelizable will be
+parallelized automatically.
+
+The build will generate the following files:
+
+- U-Boot
+    - `build/u-boot-with-spl-pbl.bin`
+- OP-TEE OS and NXP PPA
+    - `build/ppa.itb`
+- HAB Signature Data (if enabled)
+  - build/hdr_spl.out
+
+### Flashing Firmware
+
+To flash the newly built firmware, place the files listed above in the root
+directory of a FAT-formatted MicroSD card. Then, boot into recovery U-Boot as
+follows:
+
+1. Connect the Grapeboard to your computer;
+2. Establish a `minicom` connection as outlined above;
+3. Press and hold the push-button labelled `S2` on the board;
+4. Power up the board, or, if already powered up, reset it by pressing and
+releasing the push-button labelled `S1`.
+5. Release the `S2` button when U-Boot prompts you to.
+
+This should leave you at the recovery U-Boot prompt.
+
+Flash the firmware by issuing the following commands:
+
+```
+# Update U-Boot
+mmc rescan
+fatload mmc 0:1 $load_addr u-boot-with-spl-pbl.bin
+sf probe 0:0
+sf erase u-boot 200000
+sf write $load_addr u-boot $filesize
+
+# Update OPTEE-OS + PPA
+mmc rescan
+fatload mmc 0:1 $load_addr ppa.itb
+sf probe 0:0
+sf erase ppa 100000
+sf write $load_addr ppa $filesize
+
+# Update CSF Header (only if HAB is enabled)
+mmc rescan
+fatload mmc 0:1 $load_addr hdr_spl.out
+sf probe 0:0
+sf erase u-boot_hdr 40000
+sf write $load_addr u-boot_hdr $filesize
+```
+
+**Note:** Do not copy-paste more than one command at a time to the serial
+console.
+
+To reboot, type:
+
+```
+reset
+```
+
+Upon reboot, you should see messages similar to the following with HAB disabled:
+
+```
+U-Boot SPL 2018.09-g8947717e16 (Oct 21 2019 - 17:01:00 -0700)
+PPA Firmware: Version LSDK-18.09
+SEC Firmware: 'loadables' present in config
+loadables: 'trustedOS@1'
+can't get CSF - HAB disabled
+SSM not in secure/trusted state: 0x9
+Security state failure
+Continuing with non-secret testing identity
+I/TC:
+I/TC: OP-TEE version: v0.4.0-1123-gd1634ce8 #1 Tue Oct 22 00:01:37 UTC 2019 aarch64
+I/TC: Successfully captured Cyres certificate chain
+I/TC: Successfully captured Cyres private key
+I/TC: Initialized
+Trying to boot from RAM
+```
+
+If HAB is enabled, the messages will instead look as follows:
+
+```
+U-Boot SPL 2018.09-00480-gdc28a9fa63-dirty (Jan 17 2019 - 11:17:15 -0800)
+PPA Firmware: Version LSDK-18.09
+SEC Firmware: 'loadables' present in config
+loadables: 'trustedOS@1'
+I/TC:
+I/TC: OP-TEE version: v0.4.0-443-g9cdcf55b-dev #6 Sat Jan 26 05:59:52 UTC 2019 aarch64
+I/TC: Successfully captured Cyres certificate chain
+I/TC: Successfully captured Cyres private key
+I/TC: Initialized
+Trying to boot from RAM
+```
+
+### Recovery
+
+If upon reset the board fails to boot, you can repeat the procedure to re-enter
+recovery U-Boot to flash the firmware again. Flashing the firmware does not
+overwrite copy of U-Boot, which is part of the recovery ROM.
```

Comment:
grammar: "overwrite copy" -> "overwrite the copy" (insert the)

Reply:
Added.